src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -2075,6 +2156,25 @@ rd_kafka_conf_t *common_conf_setup (rd_kafka_type_t ktype,
 	if (h->stats_cb)
 		rd_kafka_conf_set_stats_cb(conf, stats_cb);
 
+	if (h->partitioner_cb) {
+		/* 
+		 * Line 2164 to line 2170 is only for testing ,convert the
+		 * PyObject h->partitioner_cb to string and print.
+		 */               
+		PyObject *ks;
+		PyObject *ks8 = NULL;
+		PyObject *vs = NULL, *vs8 = NULL;
+		const char *k = NULL;
+		ks = cfl_PyObject_Unistr(h->partitioner_cb);
+		k = cfl_PyUnistr_AsUTF8(ks, &ks8);
+		printf("Jing Liu Producer_produce partitioner_cb %s\n", k);
+
+		tconf = rd_kafka_topic_conf_new();
+		rd_kafka_topic_conf_set_partitioner_cb(tconf, partitioner_cb);
+		rd_kafka_topic_conf_set_opaque(tconf, h->partitioner_cb);
+		rd_kafka_conf_set_default_topic_conf(conf, tconf);
```

Comment:
Instead of overwriting the default topic config, which will also overwrite any previously set topic properties, such as `acks`, it is better to retrieve the default topic conf and set the partitioner on it.